CoreFoundation/RunLoop.subproj/CFSocket.c

```diff
@@ -230,9 +247,11 @@ CF_INLINE Boolean __CFSocketFdSet(CFSocketNativeHandle sock, CFMutableDataRef fd
         } else {
             fds_bits = (fd_mask *)CFDataGetMutableBytePtr(fdSet);
         }
-        if (!FD_ISSET(sock, (fd_set *)fds_bits)) {
+        fds = (fd_set *)fds_bits;
+#endif
+        if (!FD_ISSET(sock, fds)) {
             retval = true;
-            FD_SET(sock, (fd_set *)fds_bits);
+            FD_SET(sock, fds);
```

Comment:
The macOS code made sure to resize so knows that this will succeed. However the Windows code is capping at 64 sockets and does not handle failure.

I think we should avoid using `FD_SET` on Windows because it has to assume an array size of 64, but `fd_set` is designed such that the allocation can make the trailing array bigger than that if it needs to.

Reply:
Resolved per convo about large FD_SETSIZE override, although we could add an assert.